earth_enterprise/SConstruct

```diff
@@ -24,6 +26,26 @@ from khEnvironment import khEnvironment
 
 EnsureSConsVersion(0, 96, 92)
 
+# Check for the presence of the passed argument on
+# the command line.  If exists return a formated
+# string that can be passed to Gradle else return
+# an empty string
+def check_for_argument( argument ) :
+  result = ""
+  if int( ARGUMENTS.get( argument, '0' ) ) > 0 :
+    result = "-P" + argument
+  return result
+
+# Check for command line arguments, if they are found
+# pass them on to Gradle.
+def gradle_options() :
+  options = "-Pdebug"
```

Comment:
Ran into a bug here. `options` is completely overwritten on every call, so you only get the result from `check_for_argument('release')`. This needs to be updated so that it will only change `options` if a non-empty string is returned from `check_for_argument`.

Reply:
I rewrote the function like this and it seemed to work:
```
def gradle_options() :
  options = "-Pdebug"
  for opt in ['profile', 'optimize', 'release']:
    gradleOpt = check_for_argument(opt)
    if gradleOpt:
      options = gradleOpt
  return options
```